src/condor_userlog/userlog.cpp

```diff
@@ -190,42 +190,37 @@ bool clusterProcLessThan(const JobStatistics *lhs, const JobStatistics *rhs) {
 void
 display_stats()
 {
-	int i;
+	int i = 0;
 
 	// display HostStatistics
 	printf("%-15.15s %9.9s %9.9s %9.9s %9.9s %9.9s %7.7s %6.6s\n",
 		   "Host/Job", "Wall Time", "Good Time", "CPU Usage",
 		   "Avg Alloc", "Avg Lost", "Goodput", "Util.");
 	printf("\n");
 
-	HostStatistics *hs;
-	HStats.startIterations();
-	while (HStats.iterate(hs) == 1) {
-		printf("%-15.15s ", hs->host);
-		printf("%9.9s ", format_time_nosecs(hs->wall_time));
-		printf("%9.9s ", format_time_nosecs(hs->good_time));
-		printf("%9.9s ", format_time_nosecs(hs->cpu_usage));
-		printf("%9.9s ", format_time_nosecs(hs->wall_time/hs->allocations));
-		printf("%9.9s ", hs->kills ?
-			   format_time_nosecs((hs->wall_time-hs->good_time)/hs->kills) :
+	for (const auto& [key, hs]: HStats) {
+		printf("%-15.15s ", hs.host);
+		printf("%9.9s ", format_time_nosecs(hs.wall_time));
+		printf("%9.9s ", format_time_nosecs(hs.good_time));
+		printf("%9.9s ", format_time_nosecs(hs.cpu_usage));
+		printf("%9.9s ", format_time_nosecs(hs.wall_time/hs.allocations));
+		printf("%9.9s ", hs.kills ?
+			   format_time_nosecs((hs.wall_time-hs.good_time)/hs.kills) :
 			   "0+00:00");
 		printf("%6.1f%% %5.1f%%\n",
-			   hs->wall_time ?
-			   float(hs->good_time)/float(hs->wall_time)*100 : 0.0,
-			   hs->good_time ?
-			   float(hs->cpu_usage)/float(hs->good_time)*100 : 0.0);
-		delete hs;
+			   hs.wall_time ?
+			   float(hs.good_time)/float(hs.wall_time)*100 : 0.0,
+			   hs.good_time ?
+			   float(hs.cpu_usage)/float(hs.good_time)*100 : 0.0);
```

Comment:
Do we really want to cast to float here, or double?  Using 32 bit float always makes me nervous.